### PR TITLE
api: add hardcoded versioning support

### PR DESCRIPTION
**Pull request by GRISHNOV**
Add versioning support, now the module api has `_VERSION` hardcoded variable. Is part of the task [1].

1. https://github.com/tarantool/roadmap-internal/issues/204

### CODE REVIEW

File: vshard/CMakeLists.txt [hunk 1]
```diff
@@ -8,4 +8,5 @@ add_subdirectory(router)
 # Install module
 install(FILES cfg.lua error.lua consts.lua hash.lua init.lua replicaset.lua
         util.lua rlist.lua heap.lua registry.lua version.lua service_info.lua
+        vshard-version.lua
```
Comment (DifferentialOrange) [inline vshard/CMakeLists.txt:11]: Verify that deb and rpm doesn't need any additional updates.

You can use packpack for that one.
```
git clone git@github.com:tarantool/queue.git
OS=debian DIST=bullseye packpack/packpack
OS=fedora DIST=35 packpack/packpack
```

File: vshard/init.lua [hunk 2]
```diff
@@ -1,4 +1,5 @@
 return {
+    _VERSION = require('vshard.vshard-version'),
```
[the remark this answers is not in the document]
Comment (DifferentialOrange) [reply vshard/init.lua:2]: Well ,let's add a check in the CI either way.

[the remark this answers is not in the document]
Comment (GRISHNOV) [reply vshard/init.lua:2]: Done

File: .github/workflows/fast_testing.yml [hunk 3]
```diff
@@ -6,6 +6,21 @@ on:
...
+    # Skip pull request job when the source branch is in the same
+    # repository.
+    if: |
+      github.event_name == 'push' ||
+      github.event.pull_request.head.repo.full_name != github.repository
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check module version
+        # We need this step to run only on push with tag.
+        if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
```
Comment (DifferentialOrange) [inline .github/workflows/fast_testing.yml:19]: Move to the job level

File: .github/workflows/fast_testing.yml [hunk 4]
```diff
@@ -6,6 +6,20 @@ on:
   workflow_dispatch:
 
 jobs:
+  version-check:
+    # Skip pull request job when the source branch is in the same
+    # repository. Also we need this job to run only on push with tag.
+    if: |
+      (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository)
```
Comment (DifferentialOrange) [inline .github/workflows/fast_testing.yml:13]: Well. since there is no "pull request job", this one seems excessive

**Comment by DifferentialOrange** [comment]
> Add versioning support, now the module api has _VERSION hardcoded variable

Please, add a roadmap ticket link to the PR description so the motivation would be clear

File: .github/workflows/fast_testing.yml (hunk 4 above)
Comment (GRISHNOV) [reply .github/workflows/fast_testing.yml]: Done

**Review by DifferentialOrange — APPROVED** [review]

File: vshard/vshard-version.lua [hunk 5]
```diff
@@ -0,0 +1,4 @@
+-- Сontains the module version.
+-- Requires manual update in case of release commit.
+
+return '0.1.22'
```
Comment (Gerold103) [inline vshard/vshard-version.lua:4]: Please, lets not add a whole new file just for a single constant. You can either hardcode it right in `vshard/init.lua`, or add it to `vshard/consts.lua`.

**Review by Gerold103 — CHANGES_REQUESTED** [review]
Thanks for the patch!

File: vshard/vshard-version.lua (hunk 5 above)
Comment (GRISHNOV) [reply vshard/vshard-version.lua]: Moved to `vshard/consts.lua`

File: vshard/init.lua [hunk 6]
```diff
@@ -1,4 +1,5 @@
 return {
+    _VERSION = require('vshard.consts').VERSION,
```
Comment (DifferentialOrange) [inline vshard/init.lua:2]: Well, maybe it's better to require `consts` a single time

Comment (GRISHNOV) [reply vshard/init.lua]: Done

**Review by Gerold103 — APPROVED** [review]
Thanks for the patch!

**Pull request merged**